src/olympia/reviews/views.py

```diff
@@ -327,22 +327,46 @@ class ReviewViewSet(AddonChildMixin, ModelViewSet):
 
     queryset = Review.objects.all()
 
+    def set_addon_object_from_review(self, review):
+        """Set addon object on the instance from a review object."""
+        self.kwargs['addon_pk'] = str(review.addon.pk)
+        del self.addon_object  # Force get_addon_object() to try again.
```

Comment:
I wonder if we should always set `addon_object` explicitly to `None` or is that another possible state compared to the object not being set on the instance?